gravitee-am-certificate/gravitee-am-certificate-api/src/main/java/io/gravitee/am/certificate/api/ConfigurationCertUtils.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.am.certificate.api;
+
+import com.nimbusds.jose.util.JSONObjectUtils;
+import lombok.extern.slf4j.Slf4j;
+
+import java.text.ParseException;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author Lukasz GAWEL (lukasz.gawel at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+
+@Slf4j
+public final class ConfigurationCertUtils {
+
+    public static List<String> extractUsageFromCertConfiguration(String configuration) {
+        if (configuration == null) {
+            return List.of();
+        }
+        try {
+            Map<String, Object> cfg = JSONObjectUtils.parse(configuration);
+            List<String> uses = JSONObjectUtils.getStringList(cfg, "use");
+            return uses == null ? List.of() : uses;
+        } catch (ParseException e) {
+            log.debug("Problem at parsing certificate configuration", e);
```

Comment:
**question**: should it be a WARN ? I'm OK to return an empty list but having a parsing exception here is maybe something we whould be worried as it should no happen. wdyt ?

Reply:
You're right, I've changed to warn.